flow/src/main/java/com/sequenceiq/flow/service/FlowService.java

```diff
@@ -260,4 +265,35 @@ public Optional<OperationFlowsView> getLastFlowOperationByResourceCrn(String res
         List<FlowLog> flowLogs = flowLogDBService.getLatestFlowLogsByCrnInFlowChain(resourceCrn);
         return flowOperationStatisticsService.createOperationResponse(resourceCrn, flowLogs);
     }
+
+    public List<FlowLogResponse> getFlowLogsByIds(List<String> flowIds) {
+        LOGGER.info("Getting flow logs by flow ids list {}", flowIds);
+        return flowLogDBService.getFlowLogsByFlowIdsCreatedDesc(new HashSet<>(flowIds))
+                .stream().map(flowLog -> flowLogConverter.convert(flowLog)).collect(toList());
+    }
+
+    public List<FlowCheckResponse> getFlowChainsByChainIds(List<String> chainIds) {
+        LOGGER.info("Getting flow check responses by flow check ids list {}", chainIds);
+        Map<String, FlowChainLog> flowChainsMap = flowChainLogService.findAllByFlowChainIdInOrderByCreatedDesc(new HashSet<>(chainIds))
+                .stream().collect(Collectors.toMap(FlowChainLog::getFlowChainId, Function.identity()));
+        Set<String> flowChainIds = new HashSet<>(flowChainsMap.keySet());
+        Map<String, List<FlowLogWithoutPayload>> flowLogsByChainId = flowLogDBService
+                .getFlowLogsWithoutPayloadByFlowChainIdsCreatedDesc(flowChainIds)
+                .stream().collect(groupingBy(FlowLogWithoutPayload::getFlowChainId, LinkedHashMap::new, toList()));
+        return flowChainIds.stream().map(key -> {
+            LOGGER.info("Checking if there is an active flow based on flow chain id {}", key);
+            return convertFlowChainToFlowCheckResponse(flowChainsMap.get(key), null != flowLogsByChainId.get(key) ?
+                    flowLogsByChainId.get(key) : List.of());
+        }).collect(toList());
+    }
+
+    public FlowCheckResponse convertFlowChainToFlowCheckResponse(FlowChainLog chainLog, List<FlowLogWithoutPayload> relatedFlowLogs) {
```

Comment:
please make this private

Reply:
maybe move this to a separate class, like `FlowLogConverter`

Reply:
made it private - but moving it to a converter class isn't possible as it calls several other util methods from FlowService, and moving everything involves a lot of code changes in parts I haven't touched for this specific PR, and injecting flowLogService in the converter component. Please let me know if you really need me to do this